app/controllers/verification_code_controller.rb

```diff
@@ -0,0 +1,65 @@
+class VerificationCodeController < BaseController
+  before_action :is_intake_locked
+  before_action :setup_edit
+
+  def setup_edit
+    @contact_type = current_archived_intake.contact_preference
+    @contact_info = current_archived_intake.contact
+  end
+  def edit
+    @form = VerificationCodeForm.new(contact_info: @contact_info)
+    case current_archived_intake.contact_preference
+    when "text"
+      # @phone_number = current_archived_intake.phone_number
+      # ArchivedIntakeTextVerificationCodeJob.perform_later(
+      #   phone_number: @phone_number,
+      #   locale: I18n.locale
+      # )
+    when "email"
+      # @email_address = current_archived_intake.email_address
+      # ArchivedIntakeEmailVerificationCodeJob.perform_later(
+      #   email_address: @email_address,
+      #   locale: I18n.locale
+      # )
+    else
+      redirect_to root_path
+    end
+  end
+
+  def update
+    @form = VerificationCodeForm.new(verification_code_form_params, contact_info: current_archived_intake.contact)
+    @email_address = current_archived_intake.email_address
+
+    if @form.valid?
+      case current_archived_intake.contact_preference
+      when "text"
+        # TODO: Some kind of logging here
+      when "email"
+        # TODO: Some kind of logging here
+      end
+      current_archived_intake.reset_failed_attempts!
+      session[:code_verified] = true
+      redirect_to root_path
+    else
+      case current_archived_intake.contact_preference
+      when "text"
+        # TODO: Some kind of logging here
+      when "email"
+        # TODO: Some kind of logging here
+      end
+      current_archived_intake.increment_failed_attempts
```

Comment:
if the user comes back after 60 minutes + logs in again but gets the verification code wrong, do they get locked out immediately, or do they still have 2 tries left?

Reply:
devise resets it every hour

Reply:
in my recollection, it does not work that way -- I'd test it out locally by changing the 60 minutes to like a 1 minute

Reply:
you are right. We had it like that previously do you think we should change it?